test/foundry-tests/Tick.t.sol

```diff
@@ -0,0 +1,473 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.20;
+
+import "forge-std/console.sol";
+import {Test} from "forge-std/Test.sol";
+import {Vm} from "forge-std/Vm.sol";
+import {GasSnapshot} from "../../lib/forge-gas-snapshot/src/GasSnapshot.sol";
+import {Constants} from "./utils/Constants.sol";
+import {Pool} from "../../contracts/libraries/Pool.sol";
+
+contract TickTestTest is Test, GasSnapshot {
+    using Pool for Pool.State;
+
+    Pool.State public pool;
+
+    enum FeeAmount {
+        LOW,
+        MEDIUM,
+        HIGH
+    }
+
+    int24[3] TICK_SPACINGS = [int24(10), 60, 200];
+
+    int24 constant MIN_TICK = -887272;
+    int24 constant MAX_TICK = 887272;
+
```

Comment:
nit: recommend to have the test functions here since it will make readability and also those are public functions.